### ISSUE TITLE
[Windows] Uses default terminal profile

### ISSUE THREAD

**Issue by jacovanc** [tomrijndorp/vscode-finditfaster#76]
I use Bash as my default terminal in vscode. Hitting CTRL-Shift-J or U tries to run it in bash, which doesn't seem to work.
If I change the default to powershell it works as expected.

Ideally it would open powershell over the default if it is not expected to work on bash for windows.

**Comment by mdmelin** [comment]
I'm having the same error. Can this be handled on the FindItFaster end? If so, I'm happy to take a crack at it if someone can point me to where a terminal is requested from VSCode.

**Comment by ethansnow2012** [comment]
Same here. But I can not get it to work even after setting powershell as default.

Error message:
```
Failed to run checks before starting extension. Maybe this is helpful: Error: Command failed: powershell 
....
...

c:\Users\ethan\.vscode\extensions\tomrijndorp.find-it-faster-0.0.31\f ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : SecurityError: (:) [], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess
```

**Comment by subinoybiswas** [comment]
Tried fixing this issue in my fork, the logic works, made a `forceBashWindows` setting the user can toggle, but having difficulties in getting that state in the code, would be great if somebody can test it.
[https://github.com/subinoybiswas/vscode-finditfaster.git](url)

**Comment by subinoybiswas** [comment]
> Tried fixing this issue in my fork, the logic works, made a `forceBashWindows` setting the user can toggle, but having difficulties in getting that state in the code, would be great if somebody can test it. [My Fork](https://github.com/subinoybiswas/vscode-finditfaster.git)

Giving the changes here, as Prettier made the Diff wild;)
`  const localScript = (x: string) =>
    CFG.forceBashOnWindows && x !== "flight_check"
      ? vscode.Uri.file(
          `powershell "${
            path.join(context.extensionPath, commands.findFiles.script) + ".ps1"
          }"`
        )
      : vscode.Uri.file(
          path.join(context.extensionPath, x) +
            (os.platform() === "win32" ? ".ps1" : ".sh")
        );`

**Comment by joshmartinlumi** [comment]
Having the same issue, Bash is my default terminal (which fails to run this command) but it works on powershell.

I would be happy to run this in powershell, but I dont really want to change my default terminal with the aliases etc i have in bash.

Can we add a "what terminal to run on" dropdown?

**Comment by joshmartinlumi** [comment]
https://github.com/tomrijndorp/vscode-finditfaster/commit/e8209d61860344502fc75e63e003581e7a4455bb

The above code provides a not very elegant solution allowing users to provide a shell path. This could be expanded to use the same drop down option that vscode supplies you with for selecting your default terminal profile, but is beyond my expertise for the moment!

If no path is supplied in the setting (== '') then it just uses the default terminal.

I've never messed about with extension code before so apologies if it's going something glaringly wrong with it.

Complied using npx vsce package and using it on my machine.

TL;DR adds option to specify what shell you want to run the code in, allowing you to use a different terminal to your default profile.

**Comment by Steve-Mcl** [comment]
same issue here - I have bash as default shell and I get the following error when launching this:

`bash: c:UsersStephen.vscodeextensionstomrijndorp.find-it-faster-0.0.38find_files.ps1: command not found`

**Comment by joshmartinlumi** [comment]
> same issue here - I have bash as default shell and I get the following error when launching this:
> 
> `bash: c:UsersStephen.vscodeextensionstomrijndorp.find-it-faster-0.0.38find_files.ps1: command not found`

Just for anyone else who comes to this thread, the changes suggested here 

> [e8209d6](https://github.com/tomrijndorp/vscode-finditfaster/commit/e8209d61860344502fc75e63e003581e7a4455bb)
> 
> The above code provides a not very elegant solution allowing users to provide a shell path. This could be expanded to use the same drop down option that vscode supplies you with for selecting your default terminal profile, but is beyond my expertise for the moment!

Have been merged into the main branch and should be available at the next release, although there is no current timeline for the next release as of yet.

Closed PR for reference : https://github.com/tomrijndorp/vscode-finditfaster/pull/109

**Comment by VaskoTasev** [comment]
✅ Solution: Fixing "command not found" Error in Bash with Find It Faster Extension in VS Code
If you're using Bash as your default terminal in VS Code and experiencing errors like:

```
.vscode\extensions\tomrijndorp.find-it-faster-0.0.39\find_files.ps1: command not found
bash: grep: command not found
```

This happens because the Find It Faster extension tries to run a PowerShell script (find_files.ps1), but your default terminal (Bash) cannot interpret PowerShell syntax.

✅ Step-by-Step Fix
1. Keep Bash as Your Default Terminal
No need to change your main terminal setup in VS Code. You can continue using Git Bash or WSL as usual.

2. Tell the Extension to Use PowerShell Instead
The extension has a setting to specify the shell it uses internally. Set it to explicitly use PowerShell:

Go to VS Code Settings.

Search for:

`Find-it-faster › General: Shell Path For Terminal`

Set the value to:

`C:\Program Files\PowerShell\7\pwsh.exe`

🔸 Make sure that you have PowerShell 7 (pwsh) installed or set the PATH to your default PowerShell installation.

3. Save and Restart VS Code
Once this path is set, the extension will run its PowerShell scripts using pwsh.exe regardless of your default terminal.

4. Verify
Activate Find It Faster (e.g., search files), and it should now work as expected — no more .ps1: command not found errors.

💡 Summary
This setup allows you to:

✅ Use Bash as your default terminal in VS Code.
✅ Let the Find It Faster extension run via PowerShell as it was designed to.
✅ Avoid terminal conflicts and errors when invoking .ps1 scripts.
